File: Rotate Array.c
void rotate(int* nums, int numsSize, int k) {
    int tem;
    for(int j=0;j<k;j++)
    {
        tem=nums[numsSize-1];
        for(int i=numsSize-2;i>=0;i--)
            nums[i+1]=nums[i];
        nums[0]=tem;
    }
}

File: Rotate Array.js
var rotate = function(nums, k) {
    for (var rounds = 0; rounds < k; rounds++) {
        var last = nums[0];
        for (var i = 0; i < nums.length - 1; i++) {
            var tmp = nums[i + 1];
            nums[i + 1] = last;
            last = tmp;
        }
        nums[0] = last;
    }
};

